src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -830,14 +830,47 @@ static String sanitizeDefaultHost(String rawDefaultHost) {
     // Sort by ascending field index order. This is important for paged responses, where the first
     // repeated type is taken.
     fields.sort((f1, f2) -> f1.getIndex() - f2.getIndex());
+
+    // Mirror protoc's name conflict resolution behavior for fields.
+    // If a singular field's name equals that of a repeated field with "Count" or "List" suffixed,
+    // append the protobuf's field number to both fields' names.
+    // See:
+    // https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/java/java_context.cc#L60
```

Comment:
Sooner or later java_context.cc will evolve and it is likely that on L60 there will be something else. Please consider putting a link to a specific commit/tag.

Reply:
Done.